File: bindings/ruby/examples/polymorphism.rb
#!/usr/bin/ruby

require 'storage'


devicegraph = Storage::Devicegraph.new()

sda = Storage::Disk::create(devicegraph, "/dev/sda")

gpt = sda.create_partition_table(Storage::PtType_GPT)

sda1 = gpt.create_partition("/dev/sda1", Storage::Region.new(0, 100, 262144), Storage::PRIMARY)
sda2 = gpt.create_partition("/dev/sda2", Storage::Region.new(100, 100, 262144), Storage::PRIMARY)

print devicegraph


puts "partitions on gpt:"
gpt.partitions().each do |partition|
  puts "  #{partition}  #{partition.number()}"
end
puts


puts "descendants of sda:"
sda.descendants(false).each do |device|

  begin
    partition_table = Storage::to_partition_table(device)
    puts "  #{partition_table} is partition table"
  rescue Storage::DeviceHasWrongType
  end

  begin
    partition = Storage::to_partition(device)
    puts "  #{partition} #{partition.number()} is partition"
  rescue Storage::DeviceHasWrongType
  end

end
puts

File: bindings/python/examples/polymorphism.py
#!/usr/bin/python

from storage import *


environment = Environment(True)

storage = Storage(environment)

staging = storage.get_staging()

sda = Disk.create(staging, "/dev/sda")

gpt = sda.create_partition_table(PtType_GPT)

sda1 = gpt.create_partition("/dev/sda1", Region(0, 100, 512), PartitionType_PRIMARY)
sda2 = gpt.create_partition("/dev/sda2", Region(100, 100, 512), PartitionType_PRIMARY)

print(staging)


print("partitions on gpt:")
for partition in gpt.get_partitions():
  print("  %s %s" % (partition, partition.get_number()))
print()


print("descendants of sda:")
for device in sda.get_descendants(False):

  try:
    partition_table = to_partition_table(device)
    print("  %s is partition table" % partition_table)
  except DeviceHasWrongType:
    pass

  try:
    partition = to_partition(device)
    print("  %s %s is partition" % (partition, partition.get_number()))
  except DeviceHasWrongType:
    pass

print()

